packages/expo-image-picker/src/ImagePicker/ImagePicker.types.ts

```diff
@@ -6,13 +6,31 @@ export enum MediaTypeOptions {
   Images = 'Images',
 }
 
-export type ImageInfo = {
+export type NativeImageInfo = {
+  type: 'image';
   uri: string;
   width: number;
   height: number;
-  type?: 'image' | 'video';
+  exif?: Record<string, any>;
+  base64?: string;
 };
 
+export type NativeVideoInfo = {
+  type: 'video';
+  uri: string;
+  width: number;
+  height: number;
+  duration: number;
+};
+
+export type WebImageInfo = {
+  uri: string;
+  width: 0;
+  height: 0;
+};
+
+export type ImageInfo = NativeImageInfo | NativeVideoInfo | WebImageInfo;
```

Comment:
I would say that `WebImageInfo` is just a special case of `NativeImageInfo` where:
1. `exif` and `base64` are `undefined` (allowed in `NativeImageInfo`)
2. `type: 'image'` is missing

My questions would be:
- can't we select video on web? Should we have something like `WebVideoInfo`? Why not?
- why discern between `NativeImageInfo` and `WebImageInfo`? Couldn't we have one `ImageInfo`? (We would need to know in [`ExponentImagePicker.web.ts`](https://github.com/expo/expo/blob/d0af8b773a975abd463a61ad16ab2a9f862d9fcc/packages/expo-image-picker/src/ImagePicker/ExponentImagePicker.web.ts#L71-L76) what is the type of the picked asset, which [should be possible?](https://w3c.github.io/FileAPI/#dfn-type)).
- if we discerned between image and a video on web too, we would need to know metadata of the video, for that we should probably implement something like https://stackoverflow.com/a/2602578/1123156
- then we could have `VideoInfo`, `ImageInfo` and a `union type` of `AssetInfo = VideoInfo | ImageInfo` to which all this monologue led. 🙂

Reply:
Thank you for your kind comments.
I checked the links also.

I'd like to answer one by one.


> can't we select video on web?

It seems that `launchImageLibraryAsync` on Web accepts both images and videos.
https://github.com/expo/expo/blob/d0af8b773a975abd463a61ad16ab2a9f862d9fcc/packages/expo-image-picker/src/ImagePicker/ExponentImagePicker.web.ts#L10-L14


> Should we have something like WebVideoInfo? Why not?

I think we don't need to define `WebVideoInfo` separately because it resolves as single type regardless of the media type, as far as following the current code.
https://github.com/expo/expo/blob/d0af8b773a975abd463a61ad16ab2a9f862d9fcc/packages/expo-image-picker/src/ImagePicker/ExponentImagePicker.web.ts#L71-L76

cf. The reason why I named it Web`Image`Info here is that the exiting type is named as `Image`Info.
I'd like to re-consider the type name prevent to confuse...
(I feel `AssetInfo` you mentioned is nice)



> why discern between NativeImageInfo and WebImageInfo?

Because it can detect some beneficial type errors in coding like;
```typescript

if (result.type === 'video') {
  console.log(result.duration); // Should be OK
}


if (result.type === 'video') {
  console.log(result.base64); // Should be detected
}
```


> we could have VideoInfo, ImageInfo and a union type of AssetInfo = VideoInfo | ImageInfo

I strongly agree this kind of simple result.

And I also found that the root cause of the current wired situation is the current implementation on Web, as you mentioned.
It does not return `type`, does not return `duration` for video...

Then I would like to try to fix the code for Web in order to achieve something like `AssetInfo = VideoInfo | ImageInfo`.
(although my main motivation was enabling returned value for image to access `base64` property on native)

I would appreciate if you give me some more time.
And thank you so much for your insights again.

Reply:
cc @esamelson what do you think?